sdk/python/kfp/compiler/pipeline_spec_builder_test.py

```diff
@@ -42,7 +42,10 @@ def setUp(self):
         {
             'channel':
                 pipeline_channel.PipelineArtifactChannel(
-                    name='output1', task_name='task1', channel_type='Artifact'),
+                    name='output1',
+                    task_name='task1',
+                    channel_type='system.Artifact',
+                    schema_version='0.0.1'),
```

Comment:
What do you think about bundling the type and version together into something like '`system.Artifact@0.0.1`', so that this interface requires no change?

Reply:
This is a great idea -- will work on this.

Reply:
This change resulted in much more elegant code throughout the whole project. Thank you for suggesting this.

Reply:
>What do you think about bundling the type and version together into something like 'system.Artifact@0.0.1', so that this interface requires no change?

Good idea. Type names are pretty powerful and can store extra information like versions.
KFP's type structures (type ~= JSON dict) are even more powerful and can represent arbitrary type information: version, generic types, union types etc. See the Component specification design doc.